File: app/src/main/java/com/example/pcshutdowner/RecyclerViewAdapter.java
package com.example.pcshutdowner;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.ArrayList;

public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder> {

    private ArrayList<String> mIpAddresses ;
    private ArrayList<String> mComputerNames;
    private MainActivity mActivity;
    public RecyclerViewAdapter(ArrayList<String> ipAddresses, ArrayList<String> computerNames, MainActivity activity){
        mIpAddresses = ipAddresses;
        mComputerNames = computerNames;
        mActivity = activity;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_listitem,parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.parentLayout.setOnClickListener(v -> {
            mActivity.connect(mIpAddresses.get(position));
        });
        holder.ipText.setText(mIpAddresses.get(position));
        holder.computerNameText.setText(mComputerNames.get(position));

    }

    @Override
    public int getItemCount() {
        return mIpAddresses.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{
        TextView ipText;
        TextView computerNameText;
        ImageView image;
        RelativeLayout parentLayout;

        public ViewHolder(View itemView){
            super(itemView);
            image =  itemView.findViewById(R.id.computerImage);
            ipText = itemView.findViewById(R.id.ipAddressText);
            computerNameText = itemView.findViewById(R.id.computerNameText);
            parentLayout = itemView.findViewById(R.id.parent_layout);

        }
    }
}

File: app/src/main/java/com/example/pcshutdowner/RecyclerViewAdapter.kt
package com.example.pcshutdowner

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import java.util.*
import kotlin.collections.ArrayList

class RecyclerViewAdapter(
    private val computers: ArrayList<Computer>,
    private val mActivity: MainActivity
) : RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder>() {
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.layout_listitem, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(
        holder: ViewHolder,
        position: Int
    ) {
        holder.update(computers[position], mActivity)
    }

    override fun getItemCount(): Int {
        return computers.size
    }

    inner class ViewHolder(itemView: View) :
        RecyclerView.ViewHolder(itemView) {
        private var ipText: TextView = itemView.findViewById(R.id.ipAddressText)
        private var image: ImageView = itemView.findViewById(R.id.computerImage)
        private var computerNameText: TextView = itemView.findViewById(R.id.computerNameText)
        private var parentLayout: RelativeLayout = itemView.findViewById(R.id.parent_layout)

        fun update(computer: Computer, activity: MainActivity) {
            parentLayout.setOnClickListener { v: View? ->
                mActivity.connect(
                    computers[position].ip
                )
            }
            ipText.text = computers[position].ip
            computerNameText.text = computers[position].name
        }
    }

}